sips/pending/_posts/2017-02-07-make-types-behave-like-expressions.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: sip
+disqus: true
+title: SIP-NN - Make types behave like expressions 
```

Comment:
The proposal is more than infix types.
`Match infix & prefix types to meet expression rules` is better?